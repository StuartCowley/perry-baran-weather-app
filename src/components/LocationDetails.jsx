import React from "react";
import "../styles/LocationDetails.css";
import PropTypes from "prop-types";

function LocationDetails({ location, errMessage }) {
  return <h1>{errMessage || location}</h1>;
}

LocationDetails.defaultProps = {
  errMessage: "",
};

LocationDetails.propTypes = {
  errMessage: PropTypes.string,
  location: PropTypes.string.isRequired,
};

export default LocationDetails;
