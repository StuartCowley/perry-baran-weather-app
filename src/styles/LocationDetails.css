h1 {
  font-size: min(3.5rem, 9vw);
  padding-left: 5px;
}