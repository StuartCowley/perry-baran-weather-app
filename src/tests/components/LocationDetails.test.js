import React from "react";
import { render } from "@testing-library/react";
import LocationDetails from "../../components/LocationDetails";

describe("LocationDetails", () => {
  const location = "Leeds, GB";

  describe("no error message", () => {
    let screen;

    beforeEach(() => {
      screen = render(<LocationDetails location={location} />);
    });

    test("snapshot", () => {
      const { asFragment } = screen;

      expect(asFragment()).toMatchSnapshot();
    });

    test("correctly renders props", () => {
      const { getByText } = screen;

      expect(getByText(location)).toBeInstanceOf(HTMLHeadingElement);
    });
  });

  describe("error message", () => {
    test("if passed no falsey error message - renders that instead of location", () => {
      const errMessage = "Oopsies an error";
      const { queryByText } = render(
        <LocationDetails location={location} errMessage={errMessage} />
      );

      expect(queryByText(errMessage)).toBeInstanceOf(HTMLHeadingElement);
      expect(queryByText(location)).not.toBeInTheDocument();
    });
  });
});
