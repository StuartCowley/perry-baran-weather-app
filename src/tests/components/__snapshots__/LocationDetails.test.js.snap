// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationDetails no error message snapshot 1`] = `
<DocumentFragment>
  <h1>
    Leeds, GB
  </h1>
</DocumentFragment>
`;
